### ISSUE TITLE
"Given directory does not exist" from rust-analyzer

### ISSUE THREAD

**Issue by max-sixty** [binary-banter/test-each-file#6]
Possibly this is an issue with rust-analyzer, but posting here to see whether it's helpful feedback for this crate.

At https://github.com/PRQL/prql/blob/d3ed63e250b9368fbcbaa1984eef42a382a23aa5/prqlc/prqlc/tests/integration/queries.rs#L14, each of the invocations gets an error "Given directory does not exist".

It compiles & tests fine.

It's not related to the in-path module, since I tried removing that and still get the error:

<img width="498" alt="image" src="https://github.com/user-attachments/assets/2e3c72d8-bba1-4b0c-b473-b12ecf3ce69a">

<details>

```
[{
	"resource": "/Users/maximilian/workspace/prql/prqlc/prqlc/tests/integration/queries.rs",
	"owner": "rustc",
	"code": {
		"value": "macro-error",
		"target": {
			"$mid": 1,
			"external": "https://rust-analyzer.github.io/manual.html#macro-error",
			"path": "/manual.html",
			"scheme": "https",
			"authority": "rust-analyzer.github.io",
			"fragment": "macro-error"
		}
	},
	"severity": 8,
	"message": "Given directory does not exist",
	"source": "rust-analyzer",
	"startLineNumber": 52,
	"startColumn": 26,
	"endLineNumber": 52,
	"endColumn": 67
}]
```

</details>

Thanks for any guidance!

**Comment by JonathanBrouwer** [comment]
I've done some debugging and it seems that Rustc takes "prql" as the cwd (the workspace directory) when running the proc macro, while rust-analyzer takes "prql/prqlc/prqlc" as the cwd (the subcrate). So this is indeed an issue of rust-analyzer.

**Comment by max-sixty** [comment]
Thanks @JonathanBrouwer .

Do you know how the path is represented after the macro is expanded? That could help with a smaller example which I can put into rust-analyzer.

**Comment by JonathanBrouwer** [comment]
The problem is not how the macro is expanded, the problem is that the environment during execution is wrong.

I've made a simple example:
 - A workspace directory containing am empty file ROOT.txt, and two projects `test_macro` and `test_src`
 - `test_macro` defines a proc macro that during execution calls `assert!(Path::new("./ROOT.txt").is_file())`. It does not output any tokens.
 - `test_src` calls the macro defined by test_macro and does nothing else.

I've pushed the code to a branch available here:
https://github.com/binary-banter/test-each-file/tree/repro
